de-lib/src/main/java/org/iplantc/de/client/services/DiskResourceServiceFacade.java

```diff
@@ -144,29 +146,28 @@ void moveContents(final Folder sourceFolder,
      * @param callback callback executed when service call completes.
      */
     <T extends DiskResource> void deleteDiskResources(List<T> diskResources,
-                                                      AsyncCallback<HasPaths> callback);
+                                                      DECallback<HasPaths> callback);
 
     /**
      * Call service to delete disk resources in case user selects all items
      * 
      * @param selectedFolderId the folder whose contents will be deleted.
      */
-    void deleteContents(String selectedFolderId, AsyncCallback<HasPaths> callback);
+    void deleteContents(String selectedFolderId, DECallback<HasPaths> callback);
 
     /**
      * Call service to delete disk resources (i.e. {@link File}s and {@link Folder}s)
      * 
      * @param diskResources a set of <code>DiskResource</code>s to be deleted
      * @param callback callback executed when service call completes.
      */
-    void deleteDiskResources(HasPaths diskResources, AsyncCallback<HasPaths> callback);
+    void deleteDiskResources(HasPaths diskResources, DECallback<HasPaths> callback);
 
     /**
      * @param resource the <code>DiskResource</code> for which metadata will be retrieved.
      * @param callback callback executed when service call completes.
      */
-    void getDiskResourceMetaData(DiskResource resource,
-                                 AsyncCallback<DiskResourceMetadataList> callback);
+    void getDiskResourceMetaData(DiskResource resource, DECallback<DiskResourceMetadataList> callback);
```

Comment:
Hm... right now the UI is assuming there's a link between windows and services.  So the presenters decide which type of callback, in this case `DataCallback`, to use.  If that callback fails, then DECallback knows which windows to disable.  So if we have a callback that can return a 502/503 but it shouldn't disable the Data window, then it shouldn't be a `DataCallback` type. 

The `retry` button will only re-enable a window if the call that initially failed begins to succeed, because it only resends that first failing call.